### PR TITLE
Reptilian tails are now twitchy for a bit when using the "regrow tail" ability.

### PR DESCRIPTION
**Pull request by Bartimeus973**
## About the PR 
This PR adds a new critter to the game: A twitching saurian tail that splatters some blood around then dies and turns into the saurian tail organ we already had. This critter is spawned when using the "regrow tail" ability reptilians have.
The tail is colored according to the owner and lives for short bit.

Demonstration:

Tailwag3

Clearer sprite, faster speed and showing off color adaptation:

Tailwag4

Latest sprite iteration (less coily, more floppy):

tailwig3

## Why's this needed? 
Currently, if a reptilian is attacked and chooses to drop their tail as a defensive maneuver, the tail sort of lays there and doesnt attract the predator's attention. This should give saurians an edge when it comes to evasion and the element of surprise.

## Changelog

```changelog
(u)Bartimeus973
(*)Using the "regrow tail" ability as a reptilian will now make the tail twitchy for a bit!
```

### CODE REVIEW

**Comment by frawhst** [comment]
!merge_upstream

File: code/obj/critter/misc.dm [hunk 1]
```diff
@@ -1573,3 +1573,53 @@
...
+/obj/critter/livingtail
+	name = "Living tail"
+	desc = "A twitching saurian tail, you feel mildly uncomfortable looking at it."
+	icon_state = "twitchytail"
+	density = 1
+	health = 30
+	wanderer = 1
+	firevuln = 1
+	brutevuln = 1
+	butcherable = 0
```
Comment (ZeWaka) [inline code/obj/critter/misc.dm:1586]: ????

Comment (Bartimeus973) [reply code/obj/critter/misc.dm]: Removed the redundant lines

Comment (ZeWaka) [reply code/obj/critter/misc.dm]: no like why can't i butcher it

Comment (Bartimeus973) [reply code/obj/critter/misc.dm]: It despawns instantly upon dying so i didnt think that i needed to make it butcherable. Still, changed it.

Comment (ZeWaka) [reply code/obj/critter/misc.dm]: no like, why can't i butcher it into meat or something, in a feature sense
does it correctly output meat with just the flag enabled, or does the output type need to be adjusted too?

Comment (Bartimeus973) [reply code/obj/critter/misc.dm]: The meat works fine and comes as "living tail's" meat. Though realistically its impossible to get for a regular player.

File: code/obj/critter/misc.dm [hunk 2]
```diff
@@ -1573,3 +1573,50 @@
...
+			make_cleanable(/obj/decal/cleanable/blood/splatter,src.loc)
+		..()
+
+	CritterDeath()
+		if (tail_memory)
+			tail_memory.set_loc(get_turf(src))
+		else
+			new/obj/item/organ/tail/lizard(get_turf(src))
+		qdel(src)
+		..()
```
Comment (frawhst) [inline code/obj/critter/misc.dm:1622]: Should move this parent call above everything else (unless the breaks something). Deleting the tail and then calling the parent might cause weird issues when the parent call expects this to be somewhere that isn't null

**Review by frawhst — COMMENTED** [review]
Cool feature in general

File: code/obj/critter/misc.dm (hunk 2 above)
Comment (Bartimeus973) [reply code/obj/critter/misc.dm]: Done so, didnt seem to break anything.

**Review by frawhst — APPROVED** [review]

**Comment by NightmareChamillian** [comment]
I really like this flavor, but I have a few humble suggestions to throw your way regarding this feature:

-Make the sprite a little faster- it seems a little less floppy and a little more serpentine ATM
-Reduce it's health by a lot- I feel like 30 is more than a wiggly tail should have
-If a player walks into the tail or vice versa, give them a 1 second weakened effect. This way it may  possibly have some use in warding off attackers.
- If you're feeling extra spicy, have it so hostile non-drone critters, as well as space cats attack the tail- That way it would also have utility in a few circumstances.

**Comment by Bartimeus973** [comment]
Fun suggestions!
-Increased animation speed a tiny bit to make it more wriggly
-Reduce health by a third to make it less tanky, two fire extinguisher hits or three crowbars it will make it go limp
-Not sure about making it weaken people since it has collision, it already can buy you about a second since you can use it to bodyblock someone.

**Comment by Bartimeus973** [comment]
Not sure about making all hostile critters attack it, this could possibly be exploited.

However, cats now hunt the tail!

**Comment by Tarmunora** [comment]
that animation is less "twitchy tail" and more "horrible space worm monster"

**Comment by pali6** [comment]
> that animation is less "twitchy tail" and more "horrible space worm monster"

Agreed but also... is that a bad thing? 🤔

**Comment by Tarmunora** [comment]
> > that animation is less "twitchy tail" and more "horrible space worm monster"
> 
> Agreed but also... is that a bad thing? 🤔

gonna go with "yes" 🪱

**Comment by Bartimeus973** [comment]
Need i redo the entire animation or adjust the speed?

**Comment by NightmareChamillian** [comment]
honestly I feel like if you just shaved a frame or two off the beginning and end it'd seem a lot more twitchy and a lot less serpintine

**Comment by Bartimeus973** [comment]
Animation tweaked from the existing one. Added intermediary frames and made it a bit less horrifying. If this isnt good enough, let me know.

**Pull request merged**